packages/mdc-dialog/foundation.js

```diff
@@ -52,7 +52,14 @@ export default class MDCDialogFoundation extends MDCFoundation {
   constructor(adapter) {
     super(Object.assign(MDCDialogFoundation.defaultAdapter, adapter));
     this.isOpen_ = false;
-    this.componentClickHandler_ = () => this.cancel(true);
+    this.componentClickHandler_ = (evt) => {
+      if (this.adapter_.eventTargetHasClass(evt.currentTarget, cssClasses.ROOT)) {
```

Comment:
As Garbee mentioned, events should be allowed to propagate down to the document.